### ISSUE TITLE
OSIO allows user to create a project with an invalid name - results in builds failing

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#1210]
Steps to recreate:
* Create a new project / add to space
* Designate an OpenShift Project Name that begins with an integer 
* Observe the build pipeline fail with this error:

```
ERROR: Error during kubernetes apply: java.lang.RuntimeException: Failed to create Service from
pipeline.json. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at:
 https://kubernetes.default/api/v1/namespaces/ldimaggi-stage/services. Message: Service "22" is invalid: metadata.name: Invalid value: "22": a DNS-1035 label must consist of lower case alphanumeric
 characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123',
 regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?'). Received status: Status(apiVersion=v1,
 code=422, details=StatusDetails(causes=[StatusCause(field=metadata.name, message=Invalid value:
 "22": a DNS-1035 label must consist of lower case alphanumeric characters or '-', and must start and
 end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-
z0-9]*[a-z0-9])?'), reason=FieldValueInvalid, additionalProperties={})],
```

**Comment by joshuawilson** [comment]
@aslakknutsen @rawlingsj is this error/issue to do with the name set when making the space or something in openshift and k8?

**Comment by jerboaa** [comment]
This is the same or similar issue as what I've opened and got closed out without a reason: #1058

**Comment by aslakknutsen** [comment]
@joshuawilson @jerboaa It wasn't closed without a reason, but without a link. @jstrachan did a fix to the Maven plugin I believe that was suppose to address this. Not sure if this is still a problem or not?

**Comment by jerboaa** [comment]
@aslakknutsen FWIW, it's good practice to explain in a bug report why it gets closed. Be it a link or some other comment. Hence, my "without a reason" comment.

**Comment by aslakknutsen** [comment]
@jerboaa Yes, I completely agree. To be honest I'm not sure what happened in this case. I have a vauge memory about that issue being a duplicate of some other issue that the fix was linked to. But yea, this should have been linked to the duplicate at minimal.

**Comment by joshuawilson** [comment]
Fixed in #2221
